src/index.d.ts

```diff
@@ -6,7 +6,14 @@ export function props(elem: any, data?: PropData): void;
 
 export const emit: (elem: any, eventName: string, eventOptions = {}) => boolean;
 
-export class Component extends HTMLElement {}
+export class Component extends HTMLElement {
+  updatedCallback(prevProps: { [name: string]: any }): boolean;
+  renderCallback(): any | undefined;
+  renderedCallback(): void;
+  connectedCallback(): void;
+  disconnectedCallback(): void;
```

Comment:
also `renderCallback` shouldn't ever return `undefined` , the same as in preact or react. If it doest it's a programatic error